lib/Importer/ONNXModelLoader.cpp

```diff
@@ -1574,6 +1574,57 @@ Error ONNXModelLoader::loadSpaceToDepth(const ONNX_NAMESPACE::NodeProto &op,
   return Error::success();
 }
 
+Error ONNXModelLoader::loadReduceL2(const ONNX_NAMESPACE::NodeProto &op,
+                                    const ArgumentDictionaryTy &dict) {
+  const std::string &opName = loadOperatorName(op);
+  NodeValue in;
+  ASSIGN_VALUE_OR_RETURN_ERR(in, getNodeValueByName(op.input(0)));
+
+  // square
+  in = G_.createPow(opName, in, 2);
+
+  // reduceAdd
+  std::vector<unsigned_t> shapeAxes = {};
+  if (dict.count("axes")) {
+    for (int32_t axisValue : dict.at("axes")->ints()) {
+      if (axisValue < 0)
+        axisValue += in.dims().size();
+      shapeAxes.push_back((unsigned_t)axisValue);
+    }
+  } else {
+    shapeAxes.resize(in.dims().size());
+    std::iota(shapeAxes.begin(), shapeAxes.end(), 0);
+  }
+  std::sort(shapeAxes.begin(), shapeAxes.end());
+  llvm::ArrayRef<unsigned_t> axes(shapeAxes);
+  if (axes.size() > 1) {
+    auto it = std::unique(shapeAxes.begin(), shapeAxes.end());
+    if (it != shapeAxes.end())
+      RETURN_ERR("Axes values are not unique",
+                 ErrorValue::ErrorCode::MODEL_LOADER_UNSUPPORTED_SHAPE);
+  }
+  bool keepDims = true;
+  if (dict.count("keepdims")) {
+    int keepdims;
+    ASSIGN_VALUE_OR_RETURN_ERR(keepdims, loadInt(dict.at("keepdims")));
+    keepDims = (bool)keepdims;
+  }
+
+  // Reduceadd works only for single axis as of now
+  for (auto it = shapeAxes.rbegin(); it != shapeAxes.rend(); ++it) {
```

Comment:
As in other PR RE: LLVM style for `for` loops:

`for (auto it = shapeAxes.rbegin(), e = shapeAxes.rend(); it != e; ++it)`